_posts/2025-06-18-#31829.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: pr
+date: 2025-06-18
+title: "Improve TxOrphanage denial of service bounds"
+pr: 31829
+authors: [glozow]
+components: ["p2p"]
+host: glozow
+status: upcoming
+commit: 0d511965c91
+---
+
+## Notes
+
+- The [PR description](https://github.com/bitcoin/bitcoin/pull/31829#issue-2840961349) summarizes the motivations for
+  this change and the new eviction strategy: we must ensure the orphanage is DoS-resistant, but also want to prevent any
+peer from affecting another peer's usage of orphanage space.
+
+- The PR does a few things: virtualizes the `TxOrphanage` class inherited by `TxOrphanageImpl`, implements a new
+  eviction strategy, rewrites `TxOrphanage` the data structure(s) as a single `boost::multi_index_container`.
+
+- With these PR's changes, we can now "guarantee" at least 1 ancestor package worth of orphan resolution per peer at a
```

Comment:
"at least 1 ancestor package worth of orphan resolution per peer" is a bit confusing to me. I think you're hinting at 1p1c here, right? Would this be a correct rewrite?

> each peer is guaranteed to have space for at least 1 ancestor package, allowing [`TRUC`](https://bitcoinops.org/en/topics/version-3-transaction-relay/) package relay through orphan resolution

Reply:
changed to "maximum size-package"